### PR TITLE
Closes #1863 address small bugs in `MakeWorkflowList()` run as intended

### LINKED ISSUE #1863: Bugfix: `MakeWorkflowList()` assertion can be hard to debug if yaml's aren't properly specified
## Expected Behavior
`MakeWorkflowList("kri0001")` essentially takes the information inside `kri0001.yaml` and returns it into a list object.

## Current Behavior
Because the way `MakeWorkflowList()` is currently structured with the `stopifnot()` checks happening _before_ the subsetting of the user specified workflows, the function may stop if another yaml in the same path isn't specified properly.  For example, below I've written an additional yaml denoted as `break_workflow.yaml` and put it in `inst/workflows` . 

image

Then running `MakeWorkflowList("kri0001")` breaks because of line 58 of `utils-MakeWorkflowList.R` highlighted below

image

## Possible Solution
A. Subset the specific yamls to read and keep before the check
B. Improve error messaging to indicate which file the error broke on
C. A & B 

## Steps to Reproduce
Add or edit a yaml in `inst/workflows` and comment out the meta ID line of any of the files, or dont include the meta/steps section then run `MakeWorkflowList()`

## Context (Environment)
`Dev` branch version

### PR DESCRIPTION
**Pull request by zdz2101**
## Overview
`MakeWorkflowList()` would subset the file set in `strNames` _after_  reading/scanning through all the yaml files and could error out if one file in the specified path was noncompliant.

PR does the subsetting first, then adds additional cli messaging to help user identify _which_ file may be improperly specified.

Previous code base also sets names of the elements of the returned list object as the _ID_ field of the file, this may have caused confusion if there was a typo with the file name so now that issues a warning if there's a misalignment, see the 4 renamings of the "3_reporting" files

### CODE REVIEW

**Comment by zdz2101** [comment]
``` r
workflow <- gsm::MakeWorkflowList("bounds", "workflow/3_reporting")
#> ! No workflows found.
workflow_unintended <- gsm::MakeWorkflowList("Bounds", "workflow/3_reporting")
str(workflow_unintended)
#> List of 1
#>  $ Bounds:List of 4
#>   ..$ meta :List of 4
#>   .. ..$ Type       : chr "Reporting"
#>   .. ..$ ID         : chr "Bounds"
#>   .. ..$ Description: chr "Generate reporting data model including site-, country- and study-level metadata (dfGroups), metric metadata (d"| __truncated__
#>   .. ..$ Priority   : int 2
#>   ..$ spec :List of 2
#>   .. ..$ Reporting_Results:List of 6
#>   .. .. ..$ _all       :List of 1
#>   .. .. .. ..$ required: logi TRUE
#>   .. .. ..$ Numerator  :List of 1
#>   .. .. .. ..$ type: chr "numeric"
#>   .. .. ..$ Denominator:List of 1
#>   .. .. .. ..$ type: chr "numeric"
#>   .. .. ..$ Metric     :List of 1
#>   .. .. .. ..$ type: chr "numeric"
#>   .. .. ..$ Score      :List of 1
#>   .. .. .. ..$ type: chr "numeric"
#>   .. .. ..$ Flag       :List of 1
#>   .. .. .. ..$ type: chr "numeric"
#>   .. ..$ Reporting_Metrics:List of 1
#>   .. .. ..$ _all:List of 1
#>   .. .. .. ..$ required: logi TRUE
#>   ..$ steps:List of 1
#>   .. ..$ :List of 3
#>   .. .. ..$ output: chr "Reporting_Bounds"
#>   .. .. ..$ name  : chr "MakeBounds"
#>   .. .. ..$ params:List of 2
#>   .. .. .. ..$ dfResults: chr "Reporting_Results"
#>   .. .. .. ..$ dfMetrics: chr "Reporting_Metrics"
#>   ..$ path : chr "/home/zelos.zhu/R/x86_64-pc-linux-gnu-library/4.3/gsm/workflow/3_reporting/bounds.yaml"
reprex::reprex()
#> ℹ Non-interactive session, setting `html_preview = FALSE`.
#> CLIPR_ALLOW has not been set, so clipr will not run interactively
#> Error in switch(where, expr = stringify_expression(x_expr), clipboard = ingest_clipboard(), : EXPR must be a length 1 vector
```

<sup>Created on 2024-10-07 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

File: R/util-MakeWorkflowList.R [hunk 1]
```diff
@@ -43,46 +43,68 @@ MakeWorkflowList <- function(
...
-      stopifnot(utils::hasName(workflow, "meta"))
-      stopifnot(utils::hasName(workflow, "steps"))
-      stopifnot(utils::hasName(workflow$meta, "ID"))
+      if(!utils::hasName(workflow, "meta")){
+        cli::cli_abort(c("{file_name} must contain `meta` attributes."))
+      }
+      if(!utils::hasName(workflow, "steps")){
+        cli::cli_abort(c("{file_name} must contain `steps` attributes."))
+      }
+      if(!utils::hasName(workflow$meta, "ID")){
```
Comment (samussiah) [inline R/util-MakeWorkflowList.R:85]:
```suggestion
      if(!utils::hasName(workflow$meta, "Type")){
        cli::cli_abort(c("{file_name} must contain `Type` attribute in `meta` section."))
      }
      if(!utils::hasName(workflow$meta, "ID")){
```

Comment (samussiah) [reply R/util-MakeWorkflowList.R:85]: `Type` will be relied on in the data model I/O so would be nice to require here.

**Review by lauramaxwell — APPROVED** [review]
everything is working well for me! strNames is working as expected and i'm getting the errors i would expect when portions are missing or entire workflows are missing.

**Pull request merged**